### PR TITLE
Throw InputException if 'country_code'-parameter is missing

### PR DESCRIPTION
**Pull request by hAmpzter**
### Description (*)

Made changes according to wanted result in issue. Throws error if `country_code`-parameter is missing.

### Fixed Issues (if relevant)

1. magento/graphql-ce#1041: Notice: Undefined index: country_code

### Manual testing scenarios (*)

1. Set a shipping adress without `country_code`

### Questions or comments

See comment/question in issue: https://github.com/magento/graphql-ce/issues/1041#issuecomment-548597231

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by TomashKhamlai** [comment]
Please address this problem
```
PHP Code Sniffer detected 2 violation(s): FILE: ...app/code/Magento/QuoteGraphQl/Model/Cart/QuoteAddressFactory.php ---------------------------------------------------------------------- FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 2 LINES ---------------------------------------------------------------------- 68 | WARNING | [x] Expected 1 space after closing brace; newline | | found 69 | WARNING | [ ] Expected "} else {\n"; found "}\n else | | {\n" ---------------------------------------------------------------------- PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY ---------------------------------------------------------------------- Failed asserting that 2 matches expected 0.
```

**Comment by TomashKhamlai** [comment]
You can run this command block before committing the changes in the terminal of your IDE if you want to get the same error on your local environment:
```shell
php `pwd`/vendor/phpunit/phpunit/phpunit \
    --configuration `pwd`/dev/tests/static/phpunit.xml.dist \
    `pwd`/dev/tests/static/testsuite/Magento/Test/Php;
```
It checks all modified files. So if you want to check files after committing you can simply add one empty line at the end of the file. Hope this helps.

**Comment by TomashKhamlai** [comment]
@lenaorobei, can you imagine how many "if"s we need here for any other issue? Please look at #1046

File: app/code/Magento/QuoteGraphQl/Model/Cart/QuoteAddressFactory.php [hunk 1]
```diff
@@ -61,7 +61,12 @@ public function __construct(
     public function createBasedOnInputData(array $addressInput): QuoteAddress
     {
         $addressInput['country_id'] = '';
-        if ($addressInput['country_code']) {
+        if (!isset($addressInput['country_code'])) {
+            throw new GraphQlInputException(
+                __('"Required parameter "country_code" is missing')
```
Comment (TomashKhamlai) [inline app/code/Magento/QuoteGraphQl/Model/Cart/QuoteAddressFactory.php]:
```suggestion
                __('Required parameter "country_code" is missing')
```

**Comment by TomashKhamlai** [comment]
@magento run Database Compare

File: app/code/Magento/QuoteGraphQl/Model/Cart/QuoteAddressFactory.php [hunk 2]
```diff
@@ -61,7 +61,11 @@ public function __construct(
     public function createBasedOnInputData(array $addressInput): QuoteAddress
     {
         $addressInput['country_id'] = '';
-        if ($addressInput['country_code']) {
+        if (!isset($addressInput['country_code'])) {
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/Model/Cart/QuoteAddressFactory.php]: Please leave only `isset` check on line 64. `country_id` is kept here for BC purposes.
Here is the related task https://github.com/magento/graphql-ce/issues/914

**Comment by lenaorobei** [comment]
Hi @hAmpzter. The schema looks following: 
``` graphql
input CartAddressInput {
 firstname: String!
 lastname: String!
 company: String
 street: [String!]!
 city: String!
 region: String
 postcode: String
 country_code: String!
 telephone: String!
 save_in_address_book: Boolean
}
```
`country_code` is required input parameter and the case that validation does not properly happening shows that it is a framework bug. Such validation should not happen in resolvers.

Closing this PR since the framework bug is assigned to the internal team.

**Pull request closed without merging**